### ISSUE TITLE
These labels still exist in annotations

### ISSUE THREAD

**Issue by bwang482** [heartexlabs/label-studio#4766]
I have a simple task containing a multiple-choice question, and once a choice is made, a second question would appear asking for sub-categorization. The data field really just has a 'text'. 

Below is the code for the labeling interface:
```
<View>
  <Text name="text" value="$text"/>
  <View style="box-shadow: 2px 2px 5px #999;                padding: 20px; margin-top: 2em;                border-radius: 5px;">
    <Header value="Choose a category"/>
    
    <Choices name="level-1" toName="text" choice="multiple" showInLine="true" 
             allowNested="true">
    <Choice value="A"/>
      <Choice value="B"/>
    </Choices>
    
    
  <Choices name="level-2a" toName="text" 
           choice="single" showInLine="true" allowNested="true" 
           visibleWhen="choice-selected" whenTagName="level-1" 
           whenChoiceValue="A">
    <View style="width:100%">
      <Header value="Choose a A-sub-category" />
    </View>
    <Choice value="AA" />
    <Choice value="AB" />
  </Choices>
    
  <Choices name="level-2b" toName="text" 
           choice="single" showInLine="true" allowNested="true" 
           visibleWhen="choice-selected" whenTagName="level-1" 
           whenChoiceValue="B">
    <View style="width:100%">
      <Header value="Choose a B-sub-category" />
    </View>
    <Choice value="BA" />
    <Choice value="BB" />
  </Choices>
```

However, once I complete an annotation and come back to the labeling interface, it always shows validation error:

> Validation error 5f48689b-e04c-4087-8910-c524ac366af0
> These labels still exist in annotations:
> ['A'] (1 annotations).Please add labels to tag with name="level-1".

Such a baffling error since everything looks valid. I have read the comments in[ this thread](https://github.com/HumanSignal/label-studio/issues/734) but didn't get any solution. Am I missing anything obvious here? I am running the latest version of label-studio on Windows 10. 

Ps. The exported output looks ok, and if I remove the completed annotation and come back to Settings-labeling interface, the UI Preview shows up again.

**Comment by carloalbertobono** [comment]
Having an analogous issue with version 1.8.2

This configuration:

```
<Text name="text" value="$text"/>  
    	<View >  
		<Choices name="pattern" toName="text" choice="single-radio" showInline="false" allowNested="true">  
			<Choice value="A"/> 
			<Choice value="B"/>
			</Choices> 
	</View> 
</View> 
```

Gets validated, but as soon as I insert data I get back:

```
These labels still exist in annotations or drafts:
['B'] (1 annotations)
['A'] (1 annotations).Please add labels to tag with name="pattern".

```
is there a misunderstanding from my side, or it is an undesired behaviour?

thank you and all the best
Carlo

**Comment by carloalbertobono** [comment]
turned out I had to disable allowNested="true" and then it works

**Comment by bwang482** [comment]
Yeah I now have no clue how `allowNested` works. Without adding `allowNested` below is what I get from annotations results:
```
[{'value': {'choices': ['B']},
  'id': '_1uFUyzlsp',
  'from_name': 'level-1',
  'to_name': 'text',
  'type': 'choices',
  'origin': 'manual'},
 {'value': {'choices': ['BB']},
  'id': 'RXc43VhCCB',
  'from_name': 'level-2b',
  'to_name': 'text',
  'type': 'choices',
  'origin': 'manual'}]
```

Adding `allowNested="true"` to level-1 choice, I get
```
[{'value': {'choices': [['B']]},
  'id': 'NrGreQyI7z',
  'from_name': 'level-1',
  'to_name': 'text',
  'type': 'choices',
  'origin': 'manual'},
 {'value': {'choices': ['BB']},
  'id': '3blps2dgIY',
  'from_name': 'level-2b',
  'to_name': 'text',
  'type': 'choices',
  'origin': 'manual'}]
```

And if I add `allowNested="true"` to level-2 choices as well:
```
[{'value': {'choices': [['B']]},
  'id': 'hLyLfcIFeJ',
  'from_name': 'level-1',
  'to_name': 'text',
  'type': 'choices',
  'origin': 'manual'},
 {'value': {'choices': [['BB']]},
  'id': 'Fr9BC2O-Hj',
  'from_name': 'level-2b',
  'to_name': 'text',
  'type': 'choices',
  'origin': 'manual'}]
```

What I want and I think what it should be is to have BB dict within B dict, NOT anything above.

**Comment by hyp530** [comment]
Had same issue today with latest version.

**Comment by movalex** [comment]
For anyone hitting this: the validation error comes from stale label counts cached in the projects_projectsummary table (created_labels and created_labels_drafts columns), not from the actual annotations or drafts. The summary can get out of sync when annotations are deleted, labels are renamed, or tasks are removed through the API, scripts, or direct database operations.

To verify, check what the summary thinks exists:

  
    SELECT created_labels, created_labels_drafts 
    FROM projects_projectsummary 
    WHERE project_id = YOUR_PROJECT_ID;

You'll likely see labels or counts that no longer match reality. The result is a nested JSON structure like {"tag_name": {"label_name": count}}, where tag_name is the name attribute from your labeling config (e.g. <Labels name="tag_name">).

To remove a stale label entry:

    -- Remove a specific label from the summary
    UPDATE projects_projectsummary
    SET created_labels = created_labels #- '{tag_name,label_name}'
    WHERE project_id = YOUR_PROJECT_ID;

    -- Same for drafts if needed
    UPDATE projects_projectsummary
    SET created_labels_drafts = created_labels_drafts #- '{tag_name,label_name}'
    WHERE project_id = YOUR_PROJECT_ID;
